gce: Reuse unsuccessfully provisioned volumes.

jsafrane opened the pull request:
GCE PD names generated by Kubernetes are guaranteed to be unique - they
contain name of the cluster and UID of the PVC that is behind it.
Presence of a GCE PD that has the same name as we want to provision
indicates that previous provisioning did not go well and most probably
the controller manager process was restarted in the meantime.

Kubernetes should reuse this volume and not provision a new one.

Fixes #38681

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/38702)
<!-- Reviewable:end -->

saad-ali commented on pkg/cloudprovider/providers/gce/gce.go:
```diff
@@ -2479,7 +2479,12 @@ func (gce *GCECloud) CreateDisk(name string, diskType string, zone string, sizeG
-	return gce.waitForZoneOp(createOp, zone)
+	err = gce.waitForZoneOp(createOp, zone)
+	if isGCEError(err, "alreadyExists") {
+		glog.V(2).Infof("GCE PD %q already exists, reuising", name)
```
Maybe `Warningf`? Also, `reuising` -> `reusing`

jsafrane replied:
fixed both the level and typo

jsafrane committed:
gce: Reuse unsuccessfully provisioned volumes.

GCE PD names generated by Kubernetes are guaranteed to be unique - they
contain name of the cluster and UID of the PVC that is behind it.
Presence of a GCE PD that has the same name as we want to provision
indicates that previous provisioning did not go well and most probably
the controller manager process was restarted in the meantime.

Kubernetes should reuse this volume and not provision a new one.

jingxu97 commented:
/approval

jsafrane commented:
@jingxu97 
>/approval

that did not work, please /approve

jingxu97 commented:
@approve

jsafrane commented:
I approve it by myself :-)
 /approve

childsb commented:
removing saads review since changes were made and its not showing the comment.

jsafrane commented:
@k8s-bot kops aws e2e test this
